apps/src/lib/kits/maker/connectToMakerBoard.js

```diff
@@ -1,16 +1,34 @@
 /**
  * @file Serialport scanning and board controller creation for Maker Toolkit.
  */
+import {getStore} from '../../../redux';
 import {findPortWithViableDevice} from './portScanning';
 import CircuitPlaygroundBoard from './CircuitPlaygroundBoard';
+import {
+  startConnecting,
+  reportConnected,
+  reportConnectionError,
+} from './redux';
 
 /**
  * Establish a connection to a maker board and return a controller for it.
  * @returns {Promise.<CircuitPlaygroundBoard>}}
  */
 export default function connectToMakerBoard() {
+  const store = getStore();
+  const dispatch = store.dispatch.bind(store);
+  let board;
   return Promise.resolve()
+      .then(() => dispatch(startConnecting()))
       .then(findPortWithViableDevice)
-      .then(port => new CircuitPlaygroundBoard(port))
-      .then(board => board.connect().then(() => board));
+      .then(foundPort => new CircuitPlaygroundBoard(foundPort))
+      .then(newBoard => board = newBoard)
+      .then(() => board.connect())
+      .then(() => dispatch(reportConnected()))
+      .then(() => board)
```

Comment:
This stack overflow answer gives some compelling options for cancelable promises, including the bluebird library or passing a cancellation token: http://stackoverflow.com/questions/30233302/promise-is-it-possible-to-force-cancel-a-promise/30235261#30235261